// @alfresco/js-api/src/api-legacy/legacy.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
var childAssociationBody_1 = require("../api/content-rest-api/model/childAssociationBody");
exports.AssocChildBody = childAssociationBody_1.ChildAssociationBody;
var associationBody_1 = require("../api/content-rest-api/model/associationBody");
exports.AssocTargetBody = associationBody_1.AssociationBody;
var favoriteBodyCreate_1 = require("../api/content-rest-api/model/favoriteBodyCreate");
exports.FavoriteBody = favoriteBodyCreate_1.FavoriteBodyCreate;
var searchRequest_1 = require("../api/search-rest-api/model/searchRequest");
exports.QueryBody = searchRequest_1.SearchRequest;
var pathInfo_1 = require("../api/content-rest-api/model/pathInfo");
exports.PathInfoEntity = pathInfo_1.PathInfo;
var nodeEntry_1 = require("../api/content-rest-api/model/nodeEntry");
exports.MinimalNodeEntity = nodeEntry_1.NodeEntry;
var deletedNodeEntry_1 = require("../api/content-rest-api/model/deletedNodeEntry");
exports.DeletedNodeEntity = deletedNodeEntry_1.DeletedNodeEntry;
var node_1 = require("../api/content-rest-api/model/node");
exports.MinimalNode = node_1.Node;
var node_2 = require("../api/content-rest-api/model/node");
exports.MinimalNodeEntryEntity = node_2.Node;
var pathElement_1 = require("../api/content-rest-api/model/pathElement");
exports.PathElementEntity = pathElement_1.PathElement;
var siteBodyCreate_1 = require("../api/content-rest-api/model/siteBodyCreate");
exports.SiteBody = siteBodyCreate_1.SiteBodyCreate;
var groupPaging_1 = require("../api/content-rest-api/model/groupPaging");
exports.GroupsPaging = groupPaging_1.GroupPaging;
var processInstanceQueryRepresentation_1 = require("../api/activiti-rest-api/model/processInstanceQueryRepresentation");
exports.ProcessFilterRequestRepresentation = processInstanceQueryRepresentation_1.ProcessInstanceQueryRepresentation;
var taskQueryRepresentation_1 = require("../api/activiti-rest-api/model/taskQueryRepresentation");
exports.TaskQueryRequestRepresentation = taskQueryRepresentation_1.TaskQueryRepresentation;
var contentClient_1 = require("../contentClient");
exports.EcmClient = contentClient_1.ContentClient;
var processClient_1 = require("../processClient");
exports.BpmClient = processClient_1.ProcessClient;
var siteMembershipBodyCreate_1 = require("../api/content-rest-api/model/siteMembershipBodyCreate");
exports.SiteMemberBody = siteMembershipBodyCreate_1.SiteMembershipBodyCreate;
var siteMembershipBodyUpdate_1 = require("../api/content-rest-api/model/siteMembershipBodyUpdate");
exports.SiteMemberRoleBody = siteMembershipBodyUpdate_1.SiteMembershipBodyUpdate;
var sharedLinkBodyCreate_1 = require("../api/content-rest-api/model/sharedLinkBodyCreate");
exports.SharedLinkBody = sharedLinkBodyCreate_1.SharedLinkBodyCreate;
var sharedLinkBodyEmail_1 = require("../api/content-rest-api/model/sharedLinkBodyEmail");
exports.EmailSharedLinkBody = sharedLinkBodyEmail_1.SharedLinkBodyEmail;
var renditionBodyCreate_1 = require("../api/content-rest-api/model/renditionBodyCreate");
exports.RenditionBody = renditionBodyCreate_1.RenditionBodyCreate;
var siteMembershipRequestBodyCreate_1 = require("../api/content-rest-api/model/siteMembershipRequestBodyCreate");
exports.SiteMembershipRequestBody = siteMembershipRequestBodyCreate_1.SiteMembershipRequestBodyCreate;
var favoriteSiteBodyCreate_1 = require("../api/content-rest-api/model/favoriteSiteBodyCreate");
exports.FavoriteSiteBody = favoriteSiteBodyCreate_1.FavoriteSiteBodyCreate;
var siteMembershipRequestBodyUpdate_1 = require("../api/content-rest-api/model/siteMembershipRequestBodyUpdate");
exports.SiteMembershipBody = siteMembershipRequestBodyUpdate_1.SiteMembershipRequestBodyUpdate;
var nodeBodyCreate_1 = require("../api/content-rest-api/model/nodeBodyCreate");
exports.NodeBody = nodeBodyCreate_1.NodeBodyCreate;
var nodeBodyCopy_1 = require("../api/content-rest-api/model/nodeBodyCopy");
exports.CopyBody = nodeBodyCopy_1.NodeBodyCopy;
var nodeBodyMove_1 = require("../api/content-rest-api/model/nodeBodyMove");
exports.MoveBody = nodeBodyMove_1.NodeBodyMove;
var groupBodyCreate_1 = require("../api/content-rest-api/model/groupBodyCreate");
exports.GroupBody = groupBodyCreate_1.GroupBodyCreate;
