// @alfresco/js-api/index.js
"use strict";
/*!
 * @license
 * Copyright 2018 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./src/api-legacy/legacy"));
__export(require("./src/api-legacy/content-rest-api/src/index"));
__export(require("./src/api-legacy/activiti-rest-api/src/index"));
__export(require("./src/api/content-rest-api/index"));
__export(require("./src/api/discovery-rest-api/index"));
__export(require("./src/api/gs-classification-rest-api/index"));
__export(require("./src/api/gs-core-rest-api/index"));
__export(require("./src/api/auth-rest-api/index"));
__export(require("./src/api/activiti-rest-api/index"));
__export(require("./src/api/search-rest-api/index"));
var content_api_1 = require("./src/api/content-rest-api/api/content.api");
exports.ContentApi = content_api_1.ContentApi;
var alfrescoUpload_1 = require("./src/api-legacy/alfrescoUpload");
exports.AlfrescoUpload = alfrescoUpload_1.AlfrescoUpload;
var contentAuth_1 = require("./src/authentication/contentAuth");
exports.ContentAuth = contentAuth_1.ContentAuth;
var processAuth_1 = require("./src/authentication/processAuth");
exports.ProcessAuth = processAuth_1.ProcessAuth;
var oauth2Auth_1 = require("./src/authentication/oauth2Auth");
exports.Oauth2Auth = oauth2Auth_1.Oauth2Auth;
var contentClient_1 = require("./src/contentClient");
exports.ContentClient = contentClient_1.ContentClient;
var processClient_1 = require("./src/processClient");
exports.ProcessClient = processClient_1.ProcessClient;
var storage_1 = require("./src/storage");
exports.Storage = storage_1.Storage;
var alfrescoApi_1 = require("./src/alfrescoApi");
exports.AlfrescoApi = alfrescoApi_1.AlfrescoApi;
var alfrescoApiCompatibility_1 = require("./src/alfrescoApiCompatibility");
exports.AlfrescoApiCompatibility = alfrescoApiCompatibility_1.AlfrescoApiCompatibility;
var alfrescoApiClient_1 = require("./src/alfrescoApiClient");
exports.AlfrescoApiClient = alfrescoApiClient_1.AlfrescoApiClient;
var alfrescoApiConfig_1 = require("./src/alfrescoApiConfig");
exports.AlfrescoApiConfig = alfrescoApiConfig_1.AlfrescoApiConfig;
