// @alfresco/js-api/src/api/gs-classification-rest-api/api/index.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./classificationGuides.api"));
const classificationGuides_api_1 = require("./classificationGuides.api");
__export(require("./classificationReasons.api"));
const classificationReasons_api_1 = require("./classificationReasons.api");
__export(require("./declassificationExemptions.api"));
const declassificationExemptions_api_1 = require("./declassificationExemptions.api");
__export(require("./defaultClassificationValues.api"));
const defaultClassificationValues_api_1 = require("./defaultClassificationValues.api");
__export(require("./securityControlSettings.api"));
const securityControlSettings_api_1 = require("./securityControlSettings.api");
exports.GS_CLASSIFICATION_APIS = {
    ClassificationGuidesApi: classificationGuides_api_1.ClassificationGuidesApi,
    ClassificationReasonsApi: classificationReasons_api_1.ClassificationReasonsApi,
    DeclassificationExemptionsApi: declassificationExemptions_api_1.DeclassificationExemptionsApi,
    DefaultClassificationValuesApi: defaultClassificationValues_api_1.DefaultClassificationValuesApi,
    SecurityControlSettingsApi: securityControlSettings_api_1.SecurityControlSettingsApi
};
