// @alfresco/js-api/src/api/gs-core-rest-api/api/index.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./filePlans.api"));
const filePlans_api_1 = require("./filePlans.api");
__export(require("./files.api"));
const files_api_1 = require("./files.api");
__export(require("./gsSites.api"));
const gsSites_api_1 = require("./gsSites.api");
__export(require("./recordCategories.api"));
const recordCategories_api_1 = require("./recordCategories.api");
__export(require("./recordFolders.api"));
const recordFolders_api_1 = require("./recordFolders.api");
__export(require("./records.api"));
const records_api_1 = require("./records.api");
__export(require("./transferContainers.api"));
const transferContainers_api_1 = require("./transferContainers.api");
__export(require("./transfers.api"));
const transfers_api_1 = require("./transfers.api");
__export(require("./unfiledContainers.api"));
const unfiledContainers_api_1 = require("./unfiledContainers.api");
__export(require("./unfiledRecordFolders.api"));
const unfiledRecordFolders_api_1 = require("./unfiledRecordFolders.api");
exports.GS_CORE_APIS = {
    FilePlansApi: filePlans_api_1.FilePlansApi,
    FilesApi: files_api_1.FilesApi,
    GsSitesApi: gsSites_api_1.GsSitesApi,
    RecordCategoriesApi: recordCategories_api_1.RecordCategoriesApi,
    RecordFoldersApi: recordFolders_api_1.RecordFoldersApi,
    RecordsApi: records_api_1.RecordsApi,
    TransferContainersApi: transferContainers_api_1.TransferContainersApi,
    TransfersApi: transfers_api_1.TransfersApi,
    UnfiledContainersApi: unfiledContainers_api_1.UnfiledContainersApi,
    UnfiledRecordFoldersApi: unfiledRecordFolders_api_1.UnfiledRecordFoldersApi
};
